src/main/java/pix/parser/Parser.java

```diff
@@ -0,0 +1,238 @@
+package pix.parser;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+
+import pix.command.AddCommand;
+import pix.command.Command;
+import pix.command.DeleteCommand;
+import pix.command.DoneCommand;
+import pix.command.ExitCommand;
+import pix.command.FindCommand;
+import pix.command.ListCommand;
+import pix.command.UndoCommand;
+import pix.command.UndoneCommand;
+import pix.exception.PixException;
+import pix.exception.PixInvalidDateException;
+import pix.exception.PixInvalidTaskException;
+import pix.exception.PixMissingInfoException;
+import pix.exception.PixNumberFormatException;
+import pix.exception.PixUnknownCommandException;
+import pix.task.Deadline;
+import pix.task.Event;
+import pix.task.Todo;
+
+/**
+ * Parser to parse the user commands.
+ */
+public class Parser {
+    /**
+     * Parser that parses the user input.
+     *
+     * @param command The command that the user inputted in the Ui.
+     * @return Returns the command corresponding to the user input.
+     * @throws PixException Throws PixException if the user puts in invalid input.
+     */
+    public static Command parse(String command) throws PixException {
+        String[] splitCommand = command.split(" ", 2);
+        switch (splitCommand[0]) {
+        case "bye":
+            return getExitCommand(splitCommand);
+        case "todo":
+            return getTodoCommand(splitCommand);
+        case "deadline":
+            return getDeadlineCommand(splitCommand);
+        case "event":
+            return getEventCommand(splitCommand);
+        case "list":
+            return getListCommand(splitCommand);
+        case "done":
+            return getDoneCommand(splitCommand);
+        case "delete":
+            return getDeleteCommand(splitCommand);
+        case "find":
+            return getFindCommand(splitCommand);
+        case "undone":
+            return getUndoneCommand(splitCommand);
+        case "undo":
+            return getUndoCommand(splitCommand);
+        default:
+            throw new PixUnknownCommandException();
+        }
+    }
+
+    /**
+     * Runs the exit command.
+     *
+     * @param splitCommand Command input from the user.
+     * @return Returns the command for exiting Pix.
+     * @throws PixInvalidTaskException Throws the exception if the format is incorrect.
+     */
+    private static Command getExitCommand(String[] splitCommand) throws PixInvalidTaskException {
```

Comment:
Maybe instead of getExitCommand, can use confirmExitCommand. Same with the rest.